app/controllers/smart_proxies_controller.rb

```diff
@@ -2,6 +2,7 @@ class SmartProxiesController < ApplicationController
   include Foreman::Controller::AutoCompleteSearch
 
   before_filter :find_resource, :only => [:show, :edit, :update, :refresh, :ping, :tftp_server, :destroy]
+  before_filter :find_page, :only => [:show]
```

Comment:
Or I was wondering if we can't simply add the before_filter to all UI controllers and not fail when there is no page/pagelet found. We can leave it set to `nil` and in the ERB we can just issue a WARNING into logs that extension point was expecting a page/pagelet definition.

The only drawback is we are adding one hash lookup for each UI request but that's not big thing.